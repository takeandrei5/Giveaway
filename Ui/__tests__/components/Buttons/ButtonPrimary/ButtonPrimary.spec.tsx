import { ButtonPrimary } from '@components';
import { render } from '@testing-library/react';

describe('ButtonPrimary', () => {
	it('should match snapshot', () => {
		// Act
		const component = render(<ButtonPrimary />);
		// Assert
		expect(component).toMatchSnapshot();
	});

	it('should execute onClick correctly when button is clicked', () => {
		// Arrange
		const onClick = jest.fn(() => {});

		// Act
		const { getByTestId } = render(<ButtonPrimary onClick={onClick} />);
		getByTestId('button-primary').click();

		// Assert
		expect(onClick).toHaveBeenCalledTimes(1);
	});
});
