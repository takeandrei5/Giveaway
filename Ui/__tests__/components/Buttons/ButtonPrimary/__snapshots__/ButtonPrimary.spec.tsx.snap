// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPrimary should match snapshot 1`] = `
<div>
  <button
    class="chakra-button css-1k5bbcq"
    data-testid="button-primary"
    type="button"
  />
</div>
`;
